spec/controllers/internal/admin/print_controller_spec.rb

```diff
@@ -1,5 +1,71 @@
 require 'rails_helper'
 
 RSpec.describe Internal::Admin::PrintController, type: :controller do
-  it "reminds me to be implemented"
+  describe 'cover' do
+    it 'shows an alert, if no barcode is given' do
+      get :cover, barcode: ''
+      expect(response).to render_template("internal/admin/print/cover")
+      expect(flash[:alert]).to be_present
+      expect(flash[:alert]).to eq('Kein Ordner-Exemplar angegeben.')
+    end
+
+    it 'shows an alert, if no folder_instance with the given barcode exists' do
+      get :cover, barcode: -1
+      expect(response).to render_template("internal/admin/print/cover")
+      expect(flash[:alert]).to be_present
+      expect(flash[:alert]).to eq('Ordner-Exemplar mit Barcode `-1` nicht gefunden.')
+    end
+
+    it 'renders the toc template when requested for a valid folder' do
+      folder = FactoryGirl.create(:old_folder)
+      folder_instance = FactoryGirl.create(:old_folder_instance, old_folder: folder, barcodeId: 9999)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Useless assignment to variable - `folder_instance`.](https://www.codacy.com/app/lippertsjan/klausurtool-ror/file/2546221133/issues/source?bid=3141297&fileBranchId=3285107#l21)